aten/src/ATen/core/Dict_inl.h

```diff
@@ -3,11 +3,37 @@
 #include <ATen/core/ivalue.h>
 
 namespace c10 {
+namespace detail {
+inline bool DictKeyEqualTo::operator()(const IValue& lhs, const IValue& rhs) const {
+  if (lhs.isTensor() && rhs.isTensor()) {
+    // for tensors, we compare only by identity (following how it's done in Python).
+    return lhs.is(rhs);
+  }
+  // Otherwise, we first compare by identity for efficiency, then by value (see:
+  // [container equality])
+  return _fastEqualsForContainer(lhs, rhs);
+}
+inline bool operator==(const DictImpl& lhs, const DictImpl& rhs) {
+  return lhs.elementTypes.keyType == rhs.elementTypes.keyType &&
+      lhs.elementTypes.valueType == rhs.elementTypes.valueType &&
+      lhs.dict.size() == rhs.dict.size() &&
+      // see: [container equality]
+      std::equal(
```

Comment:
two equal dictionaries are not necessarily in the same order, but can still be equals. This code seems wrong.

Reply:
I thought  our dictionaries were ordered?

Reply:
okay never mind I know what you mean now